libs/runtime/ebpf_object.c

```diff
@@ -451,6 +487,38 @@ ebpf_object_reference_by_id(
     return result;
 }
 
+_Must_inspect_result_ ebpf_result_t
+ebpf_object_pointer_by_id(ebpf_id_t id, ebpf_object_type_t object_type, _Outptr_ ebpf_core_object_t** object)
+{
+    ebpf_result_t result;
+    ebpf_id_entry_t* entry = NULL;
+    ebpf_core_object_t* found_object = NULL;
+
+    result = ebpf_hash_table_find(_ebpf_id_table, (const uint8_t*)&id, (uint8_t**)&entry);
+    if (result != EBPF_SUCCESS) {
+        goto Done;
+    }
+
+    found_object = entry->object;
+    // Skip entries that have been deleted.
+    if (found_object == NULL) {
+        result = EBPF_KEY_NOT_FOUND;
+        goto Done;
+    }
+
+    // Skip entries that are not of the requested type.
+    if (entry->type != object_type) {
+        result = EBPF_KEY_NOT_FOUND;
```

Comment:
Could we use a different error code here, to differentiate between this as the 'object deleted' case above?